xbmc/video/SetInfoTag.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "SetInfoTag.h"
+
+#include "utils/StringUtils.h"
+#include "utils/XMLUtils.h"
+
+void CSetInfoTag::Reset()
+{
+  m_title.clear();
+  m_id = -1;
+  m_overview.clear();
+}
+
+bool CSetInfoTag::Load(const TiXmlElement* element, bool append, bool prioritise)
+{
+  if (!element)
+    return false;
+  if (!append)
+    Reset();
+  ParseNative(element, prioritise);
+  return true;
+}
+
+void CSetInfoTag::ParseNative(const TiXmlElement* set, bool prioritise)
+{
+  std::string value;
+
+  if (XMLUtils::GetString(set, "title", value))
+    SetTitle(value);
+
+  if (XMLUtils::GetString(set, "overview", value))
+    SetOverview(value);
+}
+
+void CSetInfoTag::SetOverview(std::string overview)
```

Comment:
`const std::string& overview`

Reply:
The correct solution for the throw is not to pass by value i think.

Reply:
```c++
void CSetInfoTag::SetOverview(const std::string& overview)
{
  m_overview = overview;
  StringUtils::Trim(m_overview);
}
```

Reply:
Done